### PR TITLE
FRM303 OLED and mLRS-WiFi-bridge mods

### PR DESCRIPTION
**Pull request by rotorman**
Docu, illustrations and 3D print files for FRM303 OLED and mLRS-WiFi-bridge mods.

### CODE REVIEW

File: docs/FLYSKY_FRM303.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ([back to main page](../README.md))
 
-The FlySky FRM303 transmitter module is a nearly perfect option for building mLRS equipment in the 2.4 GHz frequency range. Its main drawback is its price. It also has a small MCU which limits the available mLRS features. However, no hardware tweaking is required (although mods like the OLED mod are possible).
+The FlySky FRM303 transmitter module is a nearly perfect option for building mLRS equipment in the 2.4 GHz frequency range. Its main drawback is its price. It also has a small MCU which limits the available mLRS features. However, no hardware tweaking is required (although mods, like the OLED mod and mLRS-WiFi-bridge, are possible. See bottom of the page for more about optional modifications).
```
Comment (jlpoltrack) [inline docs/FLYSKY_FRM303.md:5]: Small note - could we move these last line out of parentheses.  "no hardware tweaking is required.  Although mods..."

Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done.

File: docs/FLYSKY_FRM303.md [hunk 2]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+<a href="images/FRM303-OLED-mod_step6.jpg"><img src="images/FRM303-OLED-mod_step6.jpg" height="100px"></a> 
+<a href="images/FRM303-OLED-mod_step7.jpg"><img src="images/FRM303-OLED-mod_step7.jpg" height="100px"></a>
+
+Please do note that a std. 0.96" OLED mounting holes need to be slightly enlarged to match the M2.5 screw holed of the FRM303 case.
+
+### mLRS-WiFi-bridge addition ###
+
+With mLRS-WiFi-bridge mod, you will be able to stream serial data (e.g. MAVLink) wirelessly from the FRM303 to a WiFi capable device, such as a computer, a tablet or a smartphone and use software such as [Mission Planner](https://ardupilot.org/planner/) to evaluate telemetry data or adjust flight controller parameters on the fly.
+
+Flysky FRM303 micro-JR bay adapter is spacious enough to withhold [M5Stack ATOM Lite ESP32 module](https://shop.m5stack.com/products/atom-lite-esp32-development-kit), together with [Adafruit MPM3610 5V voltage regulator](https://www.adafruit.com/product/4739). The result is a tidy and compact solution:
```
Comment (jlpoltrack) [inline docs/FLYSKY_FRM303.md:136]: Small note, just 'hold' instead of 'withhold'. "...spacious enough to hold [M5Stack..."

**Comment by jlpoltrack** [comment]
Really good - just a few minor tweaks.

**Comment by olliw42** [comment]
also very cool!!

I do have some few first comments:
- we use capitials in titles (I think)
- I guess I would start out with a first part on the simple OLED mod I did ... IMHO it would show that one really can do it with ultra-minimal hardware modification, just 4 wires and two screws ... Risto's mod is certainly infinitely cooler, but I think a most simple mod could have some attraction ... if this finds agreement I will produce few pics
- the OLED needs to be a SSD1306 (I tried SH1106 but couldn't get it to do the split screen update)

File: docs/FLYSKY_FRM303.md [hunk 3]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+
+## Optional modifications ##
+
+### OLED display addition ###
+
+The transmitter side FRM303 can be optionally equipped with a 0.96" 128x64 pixel monochrome OLED screen with I2C bus and +3.3V supply. Current parameters can be observed and the settings can be of the TX as well as connected RX side mLRS module can be altered via 5-way joystick and OLED screen.
+
+<img src="images/FRM303-OLED-mod_finished.jpg" width="400px">
+
+To activate OLED screen on FRM303, a custom firmware build is required that defines `MLRS_FEATURE_OLED`.
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done!

File: docs/FLYSKY_FRM303.md (hunk 2 above)
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Thx, fixed.

**Comment by rotorman** [comment]
@olliw42 I implemented your changes - please feel free to add your images as you like, or even remove mine if you so desire.

**Comment by olliw42** [comment]
here a piece which kind of suggested what did come to my mind based on Risto's text

### OLED Display Addition ###

The FRM303 Tx module can be optionally equipped with a 0.96" monochrome OLED screen (128x64 pixel, SSD1306 display controller, needs I2C bus and +3.3 V power supply). With the OLED, the current link statistics data can be observed, the parameter settings of the Tx module as well as those of a connected receiver can be altered, and few more functions are available. Navigation is via the 5-way joystick. The OLED can replace the CLI and the mLRS Lua script.

To activate the OLED screen, the 'OLED' firmware version (indicated by a '-oled' label in the firmware file name) needs to be flashed into the Tx module.

The OLED mod exploits the fact that the FRM303 module has four M2.5 screw holes on the front side, where the top two holes provide easy access to the interior of the housing, and the bottom two holes exactly match the distance of the mounting holes on the OLED display (the OLED mounting holes need to be slightly enlarged however to fit M2.5 screws).

The following four pictures illustrate a most simple application (click on the picture to open a bigger version):

FRM303-OLED-mod_simple_finsihed.jpg in big

  

For a more advanced modification you can find two STL files for 3D printing a mount and a housing for the OLED display under docs/files/FRM303*.stl. The following pictures illustrate the assembly process in a couple of steps (click on the picture to open a bigger version):



... the remaining pics as needed


just what my brain is spitting out

**Comment by olliw42** [comment]
> @olliw42 I implemented your changes - please feel free to add your images as you like, or even remove mine if you so desire.

oh, two tasks not synced with a semaphore ...

**Comment by olliw42** [comment]
this could be the photo I meant ... maybe i should do a better one, but for now it may serve for illustration ...

FRM303-OLED-mod_simple_finsihed

**Comment by rotorman** [comment]
@olliw42 I implemented your further suggestions and added your pic (only one, not four though?). Anyhow made some minor changes to wordings here and there as well.

File: docs/FLYSKY_FRM303.md [hunk 4]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+<a href="images/FRM303-OLED-mod_wo_case.jpg"><img src="images/FRM303-OLED-mod_wo_case.jpg" height="150px"></a> 
+
+Optionally, you can also build a housing for the OLED screen. You can find two STL files to 3D print a mount and a housing for the OLED display under `docs/files/FRM303*.stl`.
+
+Here the additional steps illustrated (click on the picture to open a bigger version):  
+<a href="images/FRM303-OLED-mod_parts.jpg"><img src="images/FRM303-OLED-mod_parts.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step2.jpg"><img src="images/FRM303-OLED-mod_step2.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step3.jpg"><img src="images/FRM303-OLED-mod_step3.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step4.jpg"><img src="images/FRM303-OLED-mod_step4.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_finished.jpg"><img src="images/FRM303-OLED-mod_finished.jpg" height="150px"></a>
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done

File: docs/FLYSKY_FRM303.md [hunk 5]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+
+<a href="images/FRM303-OLED-mod_step1.jpg"><img src="images/FRM303-OLED-mod_step1.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step5.jpg"><img src="images/FRM303-OLED-mod_step5.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step6.jpg"><img src="images/FRM303-OLED-mod_step6.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step7.jpg"><img src="images/FRM303-OLED-mod_step7.jpg" height="150px"></a>
+<a href="images/FRM303-OLED-mod_wo_case.jpg"><img src="images/FRM303-OLED-mod_wo_case.jpg" height="150px"></a> 
+
+Optionally, you can also build a housing for the OLED screen. You can find two STL files to 3D print a mount and a housing for the OLED display under `docs/files/FRM303*.stl`.
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done, although link will be functional only after the merge.

File: docs/FLYSKY_FRM303.md [hunk 6]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+<a href="images/FRM303-OLED-mod_parts.jpg"><img src="images/FRM303-OLED-mod_parts.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step2.jpg"><img src="images/FRM303-OLED-mod_step2.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step3.jpg"><img src="images/FRM303-OLED-mod_step3.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step4.jpg"><img src="images/FRM303-OLED-mod_step4.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_finished.jpg"><img src="images/FRM303-OLED-mod_finished.jpg" height="150px"></a>
+
+
+### mLRS-WiFi-bridge Addition ###
+
+With mLRS-WiFi-bridge mod, you will be able to stream serial data (e.g. MAVLink) wirelessly from the FRM303 to a WiFi capable device, such as a computer, a tablet or a smartphone and use software such as [Mission Planner](https://ardupilot.org/planner/) to evaluate telemetry data or adjust flight controller parameters on the fly.
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done.

File: docs/FLYSKY_FRM303.md [hunk 7]
```diff
@@ -101,4 +101,49 @@ Subsequent flashing can be done in two ways:
...
+
+Flysky FRM303 micro-JR bay adapter is spacious enough to hold [M5Stack ATOM Lite ESP32 module](https://shop.m5stack.com/products/atom-lite-esp32-development-kit), together with [Adafruit MPM3610 5V voltage regulator](https://www.adafruit.com/product/4739). The result is a tidy and compact solution:
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done.

File: docs/FLYSKY_FRM303.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (olliw42) [reply docs/FLYSKY_FRM303.md]: the link points to folder images, not folder files ...

File: docs/FLYSKY_FRM303.md [hunk 8]
```diff
@@ -101,4 +101,52 @@ Subsequent flashing can be done in two ways:
...
+
+The following pictures illustrate a most simple application (click on the picture to open a bigger version):
+
+<a href="images/FRM303-OLED-mod_step1.jpg"><img src="images/FRM303-OLED-mod_step1.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step5.jpg"><img src="images/FRM303-OLED-mod_step5.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step6.jpg"><img src="images/FRM303-OLED-mod_step6.jpg" height="150px"></a> 
+<a href="images/FRM303-OLED-mod_step7.jpg"><img src="images/FRM303-OLED-mod_step7.jpg" height="150px"></a>
+<a href="images/FRM303-OLED-mod_wo_case.jpg"><img src="images/FRM303-OLED-mod_wo_case.jpg" height="150px"></a> 
+
+Optionally, you can also build a housing for the OLED screen. You can find two STL files to 3D print a mount and a housing for the OLED display under [docs/files/FRM303*.stl](https://github.com/olliw42/mLRS-docu/tree/master/docs/files)
```
Comment (olliw42) [inline docs/FLYSKY_FRM303.md:126]: it should work using local referencing [docs/files/FRM303*.stl](files), doesn't it? or maybe ./files ... it works for files, see the link on in the top of this page as example

Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Done.

**Comment by rotorman** [comment]
Good point. Updated the text to reflect this.

File: docs/FLYSKY_FRM303.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
-# mLRS Documentation: FlySky FRM303 Transmitter Module #
+# mLRS Documentation: FlySky FRM303 RF Module #
```
Comment (olliw42) [inline docs/FLYSKY_FRM303.md:1]: I am not sure I should like or not like this, Throughout the docs we settled for calling the thing which goes into the back of a radio a "transmitter module" or shortened "Tx module". You could make the argument also for the R9M. In fact, for every new occasion one caould repeat the argumentation.

That's just my opinion. Pl sort out with @jlpoltrack. I would suggest however to decide and use with some consistency.

:)

File: docs/FLYSKY_FRM303.md [hunk 10]
```diff
@@ -101,4 +101,52 @@ Subsequent flashing can be done in two ways:
     - This enables programming over the USB-C port using the DFU method
     - Launch STM32CubeProgrammer and select USB as the connection method, click connect 
     - Proceed as usual
-   
+
+
+## Optional modifications ##
```
[the remark this answers is not in the document]
Comment (rotorman) [reply docs/FLYSKY_FRM303.md]: Fixed.

File: docs/FLYSKY_FRM303.md (hunk 9 above)
Comment (rotorman) [reply docs/FLYSKY_FRM303.md:1]: I did not revert this and wait now feedback from @jlpoltrack as IMHO `Transmitter Module` is incorrect term here.

**Comment by olliw42** [comment]
except of the Main title, which I leave to you and @jlpoltrack, it not only LGTM but LfantasticTM. :):)

File: docs/FLYSKY_FRM303.md (hunk 9 above)
Comment (olliw42) [reply docs/FLYSKY_FRM303.md:1]: well, might be technically incorrect, but it appears to be the commonly used and accepted terminus...
https://www.radiomasterrc.eu/product-category/transmitter-modules/

:)

Comment (jlpoltrack) [reply docs/FLYSKY_FRM303.md:1]: My two cents would be to just call it 'FlySky FRM303 Module' like the main readme page as the 'RF' sounds a bit unnecessary to me.

image

**Comment by jlpoltrack** [comment]
Great, will merge.

**Pull request merged**